### DISCUSSION TITLE
Virtual Smart Plugs

### DISCUSSION THREAD

**Discussion by JokaDev** [tolwi/hassio-ecoflow-cloud#135] [Ideas]
Hi, I've seen the idea in some threads but I'am not sure if someone is working on that. Is there any progress in emulating a smart plug to change the AC value? Especially a solution which works locally without internet connection.

**Comment by bogdancs92** [comment]
@JokaDev  
Hi, waiting for the PWS output modulation you can try : 
https://github.com/bogdancs92/ecoflow-powerstream-nodejs

via a nodejs server (which you can install locally - no internet)
and a tuya smart plug

=> automation 
    on energy modification -> set mode=0 and output = tuya energy value...

https://YOUR_URL/KEY_URL?TOKEN_KEY=TOKEN_VALUE&KEY1=100&KEY2=0 
=== pws un output de 100 watts and grid priority

it should work :)
